src/pianoroll/view/pianorollview.h

```diff
@@ -0,0 +1,254 @@
+    double displayObjectHeight() const { return m_displayObjectHeight; }
+    void setDisplayObjectHeight(double value);
+
+    Ms::Fraction playbackPosition() { return m_playbackPosition; }
+    void setPlaybackPosition(Ms::Fraction value);
+
+    void paint(QPainter*) override;
+    void keyReleaseEvent(QKeyEvent* event) override;
+    void mousePressEvent(QMouseEvent* event) override;
+    void mouseReleaseEvent(QMouseEvent* event) override;
+    void mouseMoveEvent(QMouseEvent* event) override;
+    void hoverMoveEvent(QHoverEvent* event) override;
+
+    int wholeNoteToPixelX(Ms::Fraction tick) const { return wholeNoteToPixelX(tick.numerator() / (double)tick.denominator()); }
+    int wholeNoteToPixelX(double tick) const;
+    double pixelXToWholeNote(int pixelX) const;
+    int pitchToPixelY(double pitch) const;
+    double pixelYToPitch(int tick) const;
+
+    void finishNoteGroupDrag();
+    void selectNotes(double startTick, double endTick, double lowPitch, double highPitch, NoteSelectType selType);
+    void handleSelectionClick();
+
+signals:
+    void wholeNoteWidthChanged();
+    void noteHeightChanged();
+    void toolChanged();
+    void tupletChanged();
+    void subdivisionChanged();
+    void centerXChanged();
+    void centerYChanged();
+    void displayObjectWidthChanged();
+    void displayObjectHeightChanged();
+    void playbackPositionChanged();
+
+
+private:
+    void onNotationChanged();
+    void onCurrentNotationChanged();
+    void onSelectionChanged();
+    void updateBoundingSize();
+    QRect boundingRect(Ms::Note* note);
+    QRect boundingRect(Ms::Note* note, Ms::NoteEvent* evt);
+    QString serializeSelectedNotes();
+
+    void buildNoteData();
+    void addChord(Ms::Chord* chrd, int voice, int staffIdx);
+
+//    void drawChord(QPainter* p, Ms::Chord* chrd, int voice, bool active);
+    void drawNoteBlock(QPainter* p, NoteBlock* block);
+    void drawDraggedNotes(QPainter* painter);
+    void drawDraggedNote(QPainter* painter, Ms::Fraction startTick, Ms::Fraction frac, int pitch, int track, QColor color);
+
+    NoteBlock* pickNote(int pixX, int pixY);
+    bool intersects(NoteBlock* block, int pixX, int pixY);
+    bool intersectsPixel(NoteBlock* block, int x, int y, int width, int height);
+
+    void insertNote(int modifiers);
+    void cutChord(const QPointF& pos);
+    void eraseNote(const QPointF& pos);
+    Ms::Fraction roundDownToSubdivision(double wholeNote);
+
+    void pasteNotes(const QString& copiedNotes, Ms::Fraction pasteStartTick, Ms::Fraction lengthOffset, int pitchOffset, bool xIsOffset);
+    std::vector<Ms::Note*> addNote(Ms::Fraction startTick, Ms::Fraction duration, int pitch, int track);
+    bool cutChordRest(Ms::ChordRest* targetCr, int track, Ms::Fraction cutTick, Ms::ChordRest*& cr0, Ms::ChordRest*& cr1);
+    std::vector<Ms::Note*> getSegmentNotes(Ms::Segment* seg, int track);
+
+    static void append(std::vector<Ms::Note*>& a, std::vector<Ms::Note*>& b)
```

Comment:
Questionable whether this function belongs here, but if you make the second parameter a const ref you should avoid most of those build errors.

Reply:
I've rewritten the code to avoid this method entirely.